torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -174,6 +174,18 @@ static Value* tryMatchArgument(
       auto& ostream = err()
           << arg.formatTypeMismatchMsg(value->type()->python_str());
 
+      if (auto pt = value->type()->cast<TensorType>()) {
+        if (pt->kind() == TypeKind::TensorType && pt->isInferredType()) {
```

Comment:
If the cast to `TensorType` succeeded, then it must be true that `pt->kind() == TypeKind::TensorType`, so I think you can remove that check. It should be sufficient to check only that `pt->isInferredType()` is `true`. You can also put that on the same line as `auto pt = value...`.